src/Pulsar.Client/Auth/AuthenticationDataTls.fs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Pulsar.Client.Auth
+
+open System.IO
+open System.Security.Cryptography.X509Certificates
+open System.Security.Cryptography.X509Certificates
+open Pulsar.Client.Api
+
+type internal AuthenticationDataTls(certFile: FileInfo) =
+    inherit AuthenticationDataProvider()
+    
+    let createCertCollection() =
+        use cert = new X509Certificate2(certFile.FullName)
+        if cert.HasPrivateKey then
+            let result  = X509CertificateCollection()
+            result.Add(new X509Certificate(certFile.FullName)) |> ignore

```

Comment:
why do we create X509Certificate twice?

Reply:
X509Certificate must be put in the collection, and using X509Certificate2 we check for the presence of the private key

Reply:
X509Certificate2 can be easily casted to X509Certificate

Reply:
But after casting, authorization doesn't work.